### PR TITLE
[WIP] Capture all JS dependencies in ES module imports

### PR DESCRIPTION
**Pull request by JanCVanB**
TODO:
- [ ] Manually fix remaining lint errors
- [x] Wait for #189 to merge (currently has errors that prevent full testing of this branch)
- [ ] Fully test this branch with my local Xpra server

### CODE REVIEW

File: html5/index.html [hunk 1]
```diff
@@ -23,47 +23,14 @@
...
-
-    <script type="text/javascript" src="js/lib/jsmpeg.js"></script>
-    <script type="text/javascript" src="js/lib/broadway/Decoder.js"></script>
-    <script type="text/javascript" src="js/lib/aurora/aurora.js"></script>
-    <script type="text/javascript" src="js/lib/aurora/aurora-xpra.js"></script>
-
-    <script type="text/javascript" src="js/lib/StreamSaver.js"></script>
-    <script type="text/javascript" src="js/lib/FileSaver.js"></script>
-    <script type="text/javascript" src="js/lib/jszip.js"></script>
-    <script type="text/javascript" src="js/lib/detect-zoom.js"></script>
```
Comment (JanCVanB) [inline html5/index.html:61]: Which of these unimported `lib`s (`jsmpeg`, `aurora`, `detect-zoom`, etc.) are unused/unwanted vs. still-used/wanted for their side effects?

Comment (totaam) [reply html5/index.html:61]: We need all of those!
Why would you remove them?

File transfers and stream saving are essential: #120
`jszip` is for zlib, which I intend to remove completely.
`detect-zoom` is used in `Window.js`.

Comment (JanCVanB) [reply html5/index.html:61]: I moved most of those to import statements at the top of the relevant source files. However, for some of them like `detect-zoom` I didn't see any variable-level dependencies that warranted any import statements. However, if there are implicit dependencies like `Window.js` on `detect-zoom` then I'll definitely add those! I can't yet test this branch to find those implicit dependencies at runtime. Do the import moves make sense?

Comment (totaam) [reply html5/index.html:61]: Oh, now I see. Much nicer!

**Comment by totaam** [comment]
The libraries should be excluded from the split, ie: `mitm.html`

**Comment by JanCVanB** [comment]
> The libraries should be excluded from the split, ie: `mitm.html`

Oh, could you please explain what mitm does? And what "libraries" means in this context?

**Comment by totaam** [comment]
> Oh, could you please explain what mitm does?

I don't really understand it well, it is part of [StreamSaver.js](https://github.com/jimmywarting/StreamSaver.js) - added in ce0bc83c72e628030eab248a07134f20862397fb
It goes with `sw.js` and it should be possible to move them to `/js/lib` somehow but managing relative paths quickly becomes tricky so I didn't attempt it.
Many will deploy xpra to relative paths and we have to make sure things still work there.

> And what "libraries" means in this context?

Anything that is going to be updated independently of xpra.

**Comment by JanCVanB** [comment]
@totaam Ah, now I understand mitm, thanks. I'll revert its modifications here, so that it remains maintainable.

>The libraries should be excluded from the split

Do you mean that this PR shouldn't touch `html5/lib/js/` files? If so, then this PR is impossible, since those files currently "export" their variables/functions to the global `window` context - they either need light (maintainable) modification to become compatible with ESM import syntax or they need to be replaced entirely by npm dependencies.

**Comment by totaam** [comment]
> they either need light (maintainable) modification to become compatible with ESM import syntax or they need to be replaced entirely by npm dependencies

I would prefer the former.
Can we keep most of the formatting from their upstream source?
(to make it easier to diff with upstream)

**Pull request closed without merging**